src/coreclr/inc/corinfo.h

```diff
@@ -2112,6 +2112,21 @@ class ICorStaticInfo
         CORINFO_METHOD_HANDLE meth1Hnd,
         CORINFO_METHOD_HANDLE meth2Hnd) = 0;
 
+    //------------------------------------------------------------------------------
+    // getTypeDefinition: Get the (unconstructed) type definition from a given type handle.
+    //
+    // Arguments:
+    //    type - The input type handle
+    //
+    // Return Value:
+    //   The type handle for the (unconstructed) type definition from type.
+    //
+    // Remarks:
+    //   This is equivalent of Type.GetGenericTypeDefinition(). Given a generic type handle, it will
+    //   return the original type definition (eg. for Foo<int> it will return Foo<>). This method
+    //   should only be called when the input type is in fact a generic type.
```

Comment:
"Type definition" makes sense for non-instantiated types too. The type definition of non-generic type is the type itself. It is the common semantics used in the runtime, e.g. https://github.com/dotnet/runtime/blob/main/src/coreclr/tools/Common/TypeSystem/Common/TypeDesc.cs#L611-L614 . 

Given that this API is called `getTypeDefinition` with no other adjectives, it would make more sense for it to return the type itself for non-instantiated types.

Reply:
The implementation is calling `ClassLoader::LoadTypeDefThrowing` passing `ClassLoader::PermitUninstDefOrRef`, is it possible just the docs were wrong and the implementation was actually already doing this? 🤔

Reply:
Yes, the implementation was already doing this for the most part.

I would delete the asserts and replace them with early outs (see my suggestions).

Reply:
Oh I see. Thank you! 🙂